### PR TITLE
#76: Improve TLS compatibility documentation

### LINKED ISSUE #76: TLS
The paragraph

> If you want to connect to a cluster running Exasol 7.0.x or earlier with a recent Exasol Virtual Schema (5.0.4 or later), you should explicitly disable TLS on in the JDBC connection the Virtual Schema uses, otherwise the driver will attempt to establish a TLS connection to the server that does not support it and the connection attempt will fail.

from [Exasol SQL Dialect](https://github.com/exasol/exasol-virtual-schema/blob/main/doc/dialects/exasol.md) needs to be amended.

- DBs of versions 6.2.15+ and 7.0.10+ support TLS.
- If DB is running 6.2.0..14 or 7.0.0..9, then after the initial handshake driver will switch to legacy encryption.

End result:

- DB versions 6.2.15+ and 7.0.10+ fully support TLS.
- DB versions 6.2.0..14 or 7.0.0..9 will work with 7.1 driver even without legacyencryption=1. However, it makes sense to disable TLS while connecting to these DB versions as the actual communication will happen via legacy encryption.
- For DB versions prior to 6.2 legacyencryption=1 should be used.
- The current info on TLS in the user guide (the paragraph cited above) is somewhat wrong and misleading.
- All info is available in [EXASOL-2936](https://www.exasol.com/support/browse/EXASOL-2936).

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #76.

### CODE REVIEW

**Review by jakobbraun — APPROVED** [review]

File: doc/dialects/exasol.md [hunk 1]
```diff
@@ -133,10 +133,24 @@ With version 7.1 Exasol introduced TLS encryption on the database port. Other po
...
 
+### Exasol Server Versions and TLS Support
+
+The following Exasol Server Versions support TLS connections via JDBC:
+
+* 6.2.15 and later 6.2.x versions
+* 7.0.10 and later 7.0.x versions
+* All versions from 7.1.x on
+
+Older versions (6.2.0 &hellip; 6.2.14 and 7.0.0 &hellip; 7.0.9) automatically fall back to legacy encryption. These versions are outdated anyway. If you still have one of them running, you should still disable TLS support explicitly. This is a visual reminder that TLS won't work with those versions.
```
Comment (kaklakariada) [inline doc/dialects/exasol.md:150]:
```suggestion
Older versions (6.2.0 &hellip; 6.2.14 and 7.0.0 &hellip; 7.0.9) automatically fall back to legacy encryption. These versions are outdated anyway. If you still have one of them running, you should still disable TLS support explicitly by specifying parameter `legacyencryption=1` in the JDBC URL (see details about [supported driver properties](https://docs.exasol.com/db/latest/connect_exasol/drivers/jdbc.htm#SupportedDriverProperties)). This is a visual reminder that TLS won't work with those versions.
```

**Review by kaklakariada — DISMISSED** [review]
The proposed change is optional.

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — DISMISSED** [review]

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
         run: |
           mvn --batch-mode clean verify \
               -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
-              -DtrimStackTrace=false \
+              -DtrimStackTrace=false
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply .github/workflows/ci-build.yml]: That must have been an accident. Fixed.

**Review by pj-spoelders — APPROVED** [review]
Looks good!

File: pom.xml [hunk 3]
```diff
@@ -411,142 +189,46 @@
...
-                        </goals>
-                    </execution>
-                </executions>
+                <groupId>org.sonatype.ossindex.maven</groupId>
+                <artifactId>ossindex-maven-plugin</artifactId>
+                <configuration>
+                    <excludeVulnerabilityIds>
+                        <!-- This is a vulnerability in the transitive netty test dependency introduced by the
+                        AWS SDK and in turn by exasol-test-setup-abstraction-java. AWS notes that they handle
+                        netty in a way where this cannot be exploited. -->
```
Comment (kaklakariada) [inline pom.xml:221]:
```suggestion
                        <!-- This is a vulnerability in the transitive netty test dependency introduced by the
                        AWS SDK and in turn by exasol-test-setup-abstraction-java. AWS notes that they handle
                        netty in a way where this cannot be exploited.
                        See https://github.com/aws/aws-sdk-java-v2/issues/3263#issuecomment-1163600532 -->
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

File: doc/changes/changes_6.0.3.md [hunk 4]
```diff
@@ -1,21 +1,64 @@
 # Exasol Virtual Schema 6.0.3, released 2022-02-??
```
Comment (kaklakariada) [inline doc/changes/changes_6.0.3.md:1]:
```suggestion
# Exasol Virtual Schema 6.0.3, released 2022-07-08
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**